examples/gno.land/r/demo/disperse/disperse.gno

```diff
@@ -0,0 +1,128 @@
+package disperse
+
+import (
+	"std"
+
+	tokens "gno.land/r/demo/grc20factory"
+)
+
+// DisperseGnot parses receivers and amounts and sends out gnot
+func DisperseGnot(addresses []std.Address, coins std.Coins) {
+	coinSent := std.GetOrigSend() // get Coins sent with call
+	caller := std.GetOrigCaller() // get tx sender
+
+	// Check if caller correctly sent positive amount of coins
+	for _, coin := range coins {
+		if coin.Amount <= 0 {
+			panic(ErrNegativeCoinAmount)
+		}
+	}
+
+	// Check if number of addresses and coins match
+	if len(addresses) != len(coins) {
+		panic(ErrNumAddrValMismatch)
+	}
+
+	// Get address of Disperse realm
+	realmAddr := std.CurrentRealm().Addr()
+
+	// Get Banker
+	banker := std.GetBanker(std.BankerTypeOrigSend)
+
+	var totalAmount int64
+	for _, coin := range coins {
+		totalAmount += coin.Amount
+	}
+
+	// Check if total amount of coins sent is bigger or equal to total amount of coins to be sent
+	if totalAmount > coinSent.AmountOf("ugnot") {
+		panic(ErrWrongAmount)
+	}
+
+	// Check if the realm has zero ugnot
+	if banker.GetCoins(realmAddr).AmountOf("ugnot") != coinSent.AmountOf("ugnot") {
+		panic(ErrBalanceNotEqToAmountSent)
+	}
+
+	// Send coins
+	for i, _ := range addresses {
+		banker.SendCoins(realmAddr, addresses[i], std.Coins{coins[i]})
+	}
+
+	// Return possible leftover coins
+	for _, coin := range coinSent {
+		leftoverAmt := banker.GetCoins(realmAddr).AmountOf(coin.Denom)
+		if leftoverAmt > 0 {
+			send := std.Coins{{coin.Denom, leftoverAmt}}
+			banker.SendCoins(realmAddr, caller, send)
+		}
+	}
+}
+
+// DisperseToken disperses tokens to multiple addresses
+func DisperseToken(addresses []std.Address, coins std.Coins) {
+	caller := std.GetOrigCaller() // get tx sender
+
+	// Check if caller correctly sent positive amount of coins
+	for _, coin := range coins {
+		if coin.Amount <= 0 {
+			panic(ErrNegativeCoinAmount)
+		}
+	}
+
+	// Check if number of addresses and coins match
+	if len(addresses) != len(coins) {
+		panic(ErrNumAddrValMismatch)
+	}
+
+	// Transfer tokens into the realm
+	for _, coin := range coins {
+		tokens.TransferFrom(coin.Denom, caller, std.CurrentRealm().Addr(), uint64(coin.Amount))
+	}
+
+	// Disperse tokens
+	for i, address := range addresses {
+		for y, coin := range coins {
+			if y == i {
+				tokens.Transfer(coin.Denom, address, uint64(coin.Amount))
+			}
+		}
+	}
+}
```

Comment:
since we assume `len(addresses) == len(coins)`, you could probably optimize this:
```suggestion
	for i := 0; i < len(address); i++ {
			tokens.Transfer(coins[i].Denom, address[i], uint64(coins[i].Amount))
		}
	}
}
```